AOC3/src/main/kotlin/Day16.kt

```diff
@@ -15,19 +15,16 @@ fun main() {
     while (i <= iterations) {
         orders.forEach { order ->
             when (order[0]) {
-                's' -> spinDanceMove(Integer.parseInt(order.slice(1 until order.length)))
+                's' -> spinDanceMove(order.drop(1).toInt())
                 'x' -> {
-                    val orderExchangeKey = order.split("/")
-                    val firstPosition =
-                        Integer.parseInt(orderExchangeKey[0].slice(1 until orderExchangeKey[0].length))
-                    val secondPosition = Integer.parseInt(orderExchangeKey[1])
+                    val (firstPosition, secondPosition) = order.drop(1).split("/").map { it.toInt() }

```

Comment:
As above. Using `drop` here to get rid of `x`, then splitting on "/" since we know the configuration of the input. The `String` values have to be changed to `Int` for the calculation, hence the `map`.